docs/Usage/RemoteKernelConnection.md

```diff
@@ -46,75 +46,58 @@ jupyter notebook --port=8888
-
-If all you need is a Docker Python environment to execute your code, you can read the section [**Example Jupyter Docker Stack kernel gateway**](#example-jupyter-docker-stack-kernel-gateway) (this method uses **tini** under the hood).
-
-If you want to add a temporal Kernel Gateway (for development) to your current Docker images or need to modify an existing image to add the Kernel Gateway functionality, read the section [**Example Docker kernel gateway**](#example-docker-kernel-gateway) (this method runs the kernel gateway from an interactive console).
-
-## Example Jupyter Docker Stack kernel gateway
-
-Follow this if you only need to have a simple environment to run commands inside a Docker container and nothing more.
-
-If you need to customize a Docker image (e.g. for web development) follow the section below: [**Example Docker kernel gateway**](#example-docker-kernel-gateway).
 
 ### Dockerfile
 
-- Create a `Dockerfile` based on one of the [Jupyter Docker Stacks](https://github.com/jupyter/docker-stacks).
-- Install `jupyter_kernel_gateway` in your `Dockerfile`
+- Create a `Dockerfile` based on either one of the [Jupyter Docker Stacks](https://github.com/jupyter/docker-stacks) (recommended), or your own
+- Ensure `jupyter` and `tini` are installed, either in the image or as an additional command in the `Dockerfile`
 - Expose the gateway port, in this example it will be `8888`
-- Make the command to run be the Kernel Gateway:
+- Set the `CMD` instruction to starting a notebook server, run through an init manager such as `tini`:
 
 ```Dockerfile
+# If using your own Docker image, use the following `FROM` command syntax substituting your image name
 FROM jupyter/minimal-notebook
 
-RUN pip install jupyter_kernel_gateway
-
-EXPOSE 8888
-CMD ["jupyter", "kernelgateway", "--KernelGatewayApp.ip=0.0.0.0", "--KernelGatewayApp.port=8888"]
-```
+ADD https://github.com/krallin/tini/releases/download/v0.14.0/tini /tini
+RUN chmod +x /tini
 
-### Run Docker Container with Docker commands
+# If using your own Docker image without jupyter installed:
+# RUN pip install jupyter
 
-**Note**: alternatively, see below for `docker-compose` instructions.
-
-- Build your container:
-
-```bash
-docker build -t hydro-kernel-gateway .
-```
+ENV JUPYTER_TOKEN=my_secret_token  # you can also pass this at runtime
 
-- Run your container mapping the port of the gateway
-- Give your container a name
-
-```bash
-docker run -it --rm --name hydro-kernel-gateway -p 8888:8888 hydro-kernel-gateway
+EXPOSE 8888
+ENTRYPOINT [/tini, --]
+CMD [jupyter-notebook]
+# if running as root, you need to explicitly allow this:
+# CMD [jupyter-notebook, --alow-root]
```

Comment:
I think we should add the `--no-browser` flag and maybe even the hardcoded port number to make it clearer for people who want to expose a different port.

I don't have experience with docker (I'm unsure about the placement of the `--port` flag) but something similar to this should do the job:

```Dockerfile
CMD [jupyter-notebook, --no-browser, --port 8888]
# if running as root, you need to explicitly allow this:
# CMD [jupyter-notebook, --alow-root, --no-browser, --port 8888]
```

Reply:
Just saw this: is the correct flag `--alow-root` or `--allow-root`?

Reply:
🤦‍♂️

Reply:
🤦‍♂️